src/python/pants/backend/python/target_types.py

```diff
@@ -139,72 +140,20 @@ class PexEntryPointField(StringField):
     alias = "entry_point"
 
 
+# See `target_types_rules.py` for the `ResolvePexEntryPointRequest -> ResolvedPexEntryPoint` rule.
 @dataclass(frozen=True)
 class ResolvedPexEntryPoint:
     val: Optional[str]
 
 
 @dataclass(frozen=True)
 class ResolvePexEntryPointRequest:
-    """Determine the entry_point by looking at both `PexEntryPointField` and and the sources field.
-
-    In order of precedence, this can be calculated by:
-
-    1. The `entry_point` field having a well-formed value.
-    2. The `entry_point` using a shorthand `:my_func`, and the `sources` field being set. We
-        combine these into `path.to.module:my_func`.
-    3. The `entry_point` being left off, but `sources` defined. We will use `path.to.module`.
-
-    Users can set `entry_point='<none>'` to leave off the entry point.
```

Comment:
This is never read by end users. We're changing this code too much to be worth documenting here imo. In a followup, the rule will get a good comment walking through the 7(!) different possible formats once we add support for the `app.py` shorthand. (Will be only 5 formats once sources is deprecated).